vitrina/datasets/views.py

```diff
@@ -42,3 +47,50 @@ def get_context_data(self, **kwargs):
         }
         context_data.update(extra_context_data)
         return context_data
+
+
+class DatasetCreateView(CreateView):
+    model = Dataset
+    template_name = 'base_form.html'
+    context_object_name = 'dataset'
+    form_class = NewDatasetForm
+
+    def get(self, request, *args, **kwargs):
+        if request.user.is_authenticated and\
+                request.user.organization is not None and\
+                request.user.organization.slug == self.kwargs['slug']:
```

Comment:
You also need to check this for `post`, so better use `LoginRequiredMixin` and `PermissionRequiredMixin`.

https://docs.djangoproject.com/en/3.2/topics/auth/default/#the-permissionrequiredmixin-mixin